app/models/host.rb

```diff
@@ -23,6 +23,38 @@ class Host < ApplicationRecord
   validates :account, presence: true
 
   class << self
+    def find_or_create_hosts_by_inventory_ids(ids)
+      ids.map { |id| find_or_create_from_inventory(id) }
+    end
+
+    def find_or_create_from_inventory(host_id)
+      if (existing_host = ::Pundit.policy_scope(User.current, self)
+          .where(id: host_id)).first
+        existing_host
+      else
+        save_host_from_inventory(host_id)
+      end
+    end
+
+    def save_host_from_inventory(host_id)
+      i_host = host_in_inventory(host_id)
+
+      find_or_create_by(
+        id: i_host['id'],
+        account_id: User.current.account.id
+      ) do |host|
+        host.name = i_host['display_name']
+      end
+    end
+
+    def host_in_inventory(host_id)
+      ::HostInventoryAPI.new(
+        User.current.account,
+        ::Settings.host_inventory_url,
+        nil
+      ).inventory_host(host_id)
+    end
```

Comment:
All this logic already exists in the [InventoryServiceHelper](https://github.com/RedHatInsights/compliance-backend/blob/master/app/graphql/mutations/base_mutation.rb#L12) as part of the graphql base mutations. We should just pull it out to a common location, like a model concern (app/models/concerns/).